### PR TITLE
[FLINK-18968] Translate README.md to Chinese

### PR DESCRIPTION
**Pull request by wangzzu**
Translate README.md to Chinese. New file is named README.zh.md
Methodology: Google Translate on HTML + Manual Correction

### CODE REVIEW

**Comment by carp84** [comment]
There should be some license issue according to [the build log here](https://github.com/apache/flink-statefun/runs/1440626235?check_suite_focus=true#step:4:14143), please check locally and fix it. Thanks.

**Comment by wangzzu** [comment]
@carp84 Thanks, i have fixed it.

File: README.zh.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+`KeyedProcessFunction` 。函数 ``类型`` 等同于处理函数转换（process function transformation），而 `` ID `` 则是键值（key）。不同之处
+在于，函数不是在定义数据流（拓扑）的有向无环图（DAG）中组装，
+而是使用地址将事件任意发送到所有其他函数。
+
+#### 入口和出口
+
+* _入口_ （Ingress）是事件最初到达 Stateful Functions 应用程序的方式。
+入口可以是消息队列，日志或 HTTP 服务器 —— 任何可以产生事件并交由应用程序处理的系统。
+
+* _路由器_（Router）将入口（Ingress）与 stateful function 连接起来，以确定哪个函数实例应该在最开始时处理来自 Ingress 的事件。
```
Comment (carp84) [inline README.zh.md:64]:
```suggestion
* _路由_（Router）将入口（Ingress）与 stateful function 连接起来，以确定哪个函数实例应该在最开始时处理来自入口的事件。
```
Refer to https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications

File: README.zh.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+* _出口_（Egress）是一种以标准化方式从应用程序发送事件的方法。
+出口是可选的，也有可能没有事件需要从应用程序中发送出去，函数会完成事件的处理或直接调用其他应用程序。
+
+### <a name="modules"></a>模块(Module)
+
+_模块_（Module）是将核心构建单元添加到一个 Stateful Functions 应用程序的入口，这些核心构建单元包括：入口（Ingress）、出口（Egress）、路由器（Router）和有状态函数。
```
Comment (carp84) [inline README.zh.md:71]:
```suggestion
_模块_（Module）是将核心构建单元添加到一个 Stateful Functions 应用程序的入口，这些核心构建单元包括：入口（Ingress）、出口（Egress）、路由（Router）和有状态函数。
```

File: README.zh.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+Stateful Functions Runtime 旨在提供一组类似于 [无服务器函数](https://martinfowler.com/articles/serverless.html) 的属性，但适用于有状态的场景。
+
+<p align="center">
+  <img src="docs/fig/stateful_functions_overview-ops.png" width="600px"/>
+</p>
+
+Runtime 基于 Apache Flink<sup>®</sup> 构建，并具有以下设计原则：
+
+*  __逻辑上计算/状态共置__：消息传递，状态访问/更新和函数调用在一起紧密管理，这在抽象层面就天然地保证了一致性。
```
Comment (carp84) [inline README.zh.md:87]:
```suggestion
* __逻辑上计算/状态共置__：消息传递，状态访问/更新和函数调用在一起紧密管理，这在抽象层面就天然地保证了一致性。
```

File: README.zh.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+   * [使用 Docker 映像进行部署](#docker)
+   * [作为 Flink 作业部署](#flink)
+- [参与贡献](#contributing)
+- [开源软件许可](#license)
+
+## <a name="core-concepts"></a>核心概念
+
+### <a name="abstraction"></a>摘要
+
+Stateful Functions 应用程序由以下原语组成：有状态函数，入口(ingresses)，路由器(routers)和出口(egresses)。
```
Comment (carp84) [inline README.zh.md:34]:
```suggestion
Stateful Functions 应用程序由以下原语组成：有状态函数，入口(ingresses)，路由(routers)和出口(egresses)。
```
Refer to https://cwiki.apache.org/confluence/display/FLINK/Flink+Translation+Specifications

**Review by carp84 — CHANGES_REQUESTED** [review]
Thanks for the quick action @wangzzu . The translation overall LGTM, just some inline comments.

**Comment by wangzzu** [comment]
@carp84 thanks for your reviewing, I have fixed these on your suggestions.

**Review by carp84 — APPROVED** [review]
+1, the latest changes LGTM. Thanks for the contribution @wangzzu 

@tzulitai @klion26 could you also take a look here? I believe the Chinese translation would help China users to learn StateFun and try it out so would like to merge it asap. Thanks.

**Comment by klion26** [comment]
@carp84 thanks for the remainder, will have a look now.

File: README.zh.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+* 函数实例是 _虚拟的_ ，因为它们不总是同时在内存中活跃。在任何一个时间点，只有一小部分函数及其状态作为实际对象存在。
+当一个虚拟实例接收到消息时，将配置一个对象并带着该虚拟实例的状态加载，然后处理该消息。与虚拟内存类似，许多函数的状态可能
+在任何时间点都被“交换出去”（swap out）。
+
+* 函数的每个虚拟实例都有其自己的状态，可以通过局部变量访问，
+并且该状态是私有的，对于该实例来说是本地的。
+
+如果您知道 Apache Flink 的 DataStream API，则可以将 Stateful Functions 考虑为轻量级的
+`KeyedProcessFunction` 。函数 ``类型`` 等同于处理函数转换（process function transformation），而 `` ID `` 则是键值（key）。不同之处
+在于，函数不是在定义数据流（拓扑）的有向无环图（DAG）中组装，
```
Comment (klion26) [inline README.zh.md:56]: nit: 中文来说，不建议这一行（`在于，函数不是在定义数据流（拓扑）的有向无环图（DAG）中组装`）和上一行分开。因为他们是连着的一句话，但是渲染之后这里会有空格，看上去会比较奇怪。

其他地方也类似

File: README.zh.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+<p align="center">
+  <img src="docs/fig/stateful_functions_overview.png" width="650px"/>
+</p>
+
+#### Stateful Functions
+
+* 一个 _stateful function_ 是通过消息调用的一小段逻辑/代码。每个 stateful function 都是作为 _函数类型_ 的唯一可
+调用 _虚拟实例_ 存在。每个实例都通过其 ``type`` 以及 type 中的唯一 ``ID``（字符串）来寻址。
+
+* Stateful Functions 可以从 ingress 或任何其他的 stateful function（包括其自身）中调用，调用者只需要知道目标函数的逻辑地址即可。
```
Comment (klion26) [inline README.zh.md:45]: ingress 这里是不是也翻译比较好呢？

**Review by klion26 — APPROVED** [review]
@wangzzu thanks for the contribution. Overall the translation LGTM now just left two minor comments.

**Comment by tzulitai** [comment]
Thanks @wangzzu! I read over this and it looks good to me.
Thanks a lot to @klion26 and @carp84 for reviewing this.

@klion26 @carp84 +1 on my side, please feel free to merge this.

**Comment by klion26** [comment]
@tzulitai thanks for the review. 
@wangzzu could you please squash all your commits into two commits: one for addressing comments for the translation, the other one for the checkstyle fix (please left the commits of @billyrrr as it was, there will be four commits after squashing) then I can merge it.

**Comment by wangzzu** [comment]
@klion26 thx, the commits have been squashed.

**Comment by klion26** [comment]
@wangzzu thanks for the work, merged

**Pull request closed without merging**